### PR TITLE
Zone Integration Demo

### PR DESCRIPTION
**Pull request by suyashSax**
Zone integrations for interstitials, MRecs, banners and rewarded videos

### CODE REVIEW

File: Objective-C Demo App/iOS-SDK-Demo/ALDemoInterstitalZoneViewController.m [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+@interface ALDemoInterstitalZoneViewController ()<ALAdLoadDelegate, ALAdDisplayDelegate, ALAdVideoPlaybackDelegate>
+@property (nonatomic, strong) ALAd *ad;
+@end
+
+@implementation ALDemoInterstitalZoneViewController
+
+- (IBAction)loadInterstitial:(id)sender
+{
+    [self log: @"Interstitial loading..."];
+    [[ALSdk shared].adService loadNextAdForZoneIdentifier:@"YOUR_ZONE_ID" andNotify:self];
```
Comment (thomasmso) [inline Objective-C Demo App/iOS-SDK-Demo/ALDemoInterstitalZoneViewController.m]: AppLovin code formatting has a space character after the `:` for the method arguments.

File: Objective-C Demo App/iOS-SDK-Demo/ALDemoRewardedVideosZoneViewController.m [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+#pragma mark - View Lifecycle
+
+- (void)viewDidLoad
+{
+    [super viewDidLoad];
+    
+    // Set an optional user identifier used for S2S callbacks
+    [ALIncentivizedInterstitialAd setUserIdentifier: @"DEMO_USER_IDENTIFIER"];
+    
+    self.incentivizedInterstitial = [[ALIncentivizedInterstitialAd alloc] initWithZoneIdentifier:@"YOUR_ZONE_ID"];
```
Comment (thomasmso) [inline Objective-C Demo App/iOS-SDK-Demo/ALDemoRewardedVideosZoneViewController.m]: Formatting

File: Swift Demo App/iOS-SDK-Demo-Swift.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -62,13 +62,19 @@
 		1DDB3A071BF5815D002561A0 /* ALCarouselMediaView.m in Sources */ = {isa = PBXBuildFile; fileRef = 1DDB3A061BF5815D002561A0 /* ALCarouselMediaView.m */; };
 		1DDB3A0D1BF58272002561A0 /* ALDemoRSSFeedRetriever.m in Sources */ = {isa = PBXBuildFile; fileRef = 1DDB3A0A1BF58272002561A0 /* ALDemoRSSFeedRetriever.m */; };
 		1DDB3A0E1BF58272002561A0 /* ALDemoArticle.m in Sources */ = {isa = PBXBuildFile; fileRef = 1DDB3A0C1BF58272002561A0 /* ALDemoArticle.m */; };
+		29034E9F6B88F35DFDD40479 /* Pods_iOS_SDK_Demo_Swift.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = BF244B6EA5DE3FFA0AE2F090 /* Pods_iOS_SDK_Demo_Swift.framework */; };
```
Comment (thomasmso) [inline Swift Demo App/iOS-SDK-Demo-Swift.xcodeproj/project.pbxproj]: Can you remove all references to your local CocoaPod files from the xcode project file?

File: Swift Demo App/iOS-SDK-Demo/ALDemoMRecZoneViewController.swift [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    }
+    
+    func ad(_ ad: ALAd, didReturnToApplicationFor adView: ALAdView)
+    {
+        self.log("Banner did return to application")
+    }
+    
+    func ad(_ ad: ALAd, didFailToDisplayIn adView: ALAdView, withError code: ALAdViewDisplayErrorCode)
+    {
+        self.log("Banner did fail to display with error code: \(code)")
```
Comment (thomasmso) [inline Swift Demo App/iOS-SDK-Demo/ALDemoMRecZoneViewController.swift]: These logs all say `"Banner ...'`, when this is a view controller for MRecs.

File: Objective-C Demo App/iOS-SDK-Demo/ALDemoInterstitalZoneViewController.m [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+@property (nonatomic, strong) ALAd *ad;
+@property (nonatomic, weak) IBOutlet UIBarButtonItem *showButton;
+@end
+
+@implementation ALDemoInterstitalZoneViewController
+
+- (IBAction)loadInterstitial:(id)sender
+{
+    [self log: @"Interstitial loading..."];
+    [[ALSdk shared].adService loadNextAdForZoneIdentifier: @"YOUR_ZONE_ID" andNotify:self];
```
Comment (thomasmso) [inline Objective-C Demo App/iOS-SDK-Demo/ALDemoInterstitalZoneViewController.m]: Could we also put a space to the right of the `:` before `self`?

Comment (suyashSax) [reply Objective-C Demo App/iOS-SDK-Demo/ALDemoInterstitalZoneViewController.m]: o my b for not noticing

**Comment by thomasmso** [comment]
Cool - lets merge this after Android is ready

**Pull request merged**